assignments/this.js

```diff
@@ -1,26 +1,46 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. implicit binding - the object standing before the dot is what the keyword will be bound to
+* 2. explicit binding - you can force a function call to use of an object with the 'this' binding without using a property function on an object
+* 3. new binding - allows you to create a new object referenced by 'this' and returned at the end of the code.
+* 4. window/global - binds 'this' to the console
 *
 * write out a code example of each explanation above
 */
 
 // Principle 1
 
 // code example for Window Binding
+console.log(this);
 
 // Principle 2
 
 // code example for Implicit Binding
-
+var MyObject = function (){
```

Comment:
Implicit binding has more to do with the object and methods inside of it. For example:
```
const mage = {
   name: "merlin",
   spell: function() => {
        return `my name is ${this.name}`
   }
}

mage.spell();
```

this is bound to the mage object because it is to the left of the invokation of `spell()`